src/kakarot/execution_context.cairo

```diff
@@ -58,13 +59,16 @@ namespace ExecutionContext {
         dw 0;  // evm_contract_address
         dw 0;  // calling_context
         dw 0;  // sub_context
+        dw 0;  // block_context
     }
 
     // @notice Initialize the execution context.
     // @dev set the initial values before executing a piece of code
     // @param call_context The call context.
     // @return The initialized execution context.
-    func init(call_context: model.CallContext*) -> model.ExecutionContext* {
+    func init{range_check_ptr}(
+        call_context: model.CallContext*, block_context: model.BlockContext*
+    ) -> model.ExecutionContext* {
```

Comment:
Adding `block_context` to `init` and `init_at_address` to add block context to the ExecutionContext